iree/build_defs.oss.bzl

```diff
@@ -56,7 +56,10 @@ IREE_DRIVER_MODULES = [
     "//iree/hal/vmla:vmla_driver_module",
     "//iree/hal/vulkan:vulkan_driver_module",
     "//iree/hal/llvmjit:llvmjit_driver_module",
-]
+] + select({
+    "//iree:macos_x86_64": ["//iree/hal/metal:metal_driver_module"],
+    "//conditions:default": [],
+})
```

Comment:
<sup>(just dropping a comment here for a thread)</sup>

Oooo this (in the commit description) is neat:
```
bazel test -c opt  -k //iree/...  --deleted_packages=//iree/hal/metal
```

We could even write that `--deleted_packages=//iree/hal/metal` to `configured.bazelrc` in https://github.com/google/iree/blob/main/configure_bazel.py if the host platform isn't macos. That would let developers build stuff like `iree/...` without needing to add it explicitly.

@GMNGeoffrey how does that sound to you? Any ideas for how to work that into https://github.com/google/iree/blob/main/build_tools/bazel/build_core.sh and the other scripts used by CI?

Reply:
Oooh that is an interesting approach. Rather than doing it configured.bazelrc, maybe make it part of the configs for each platform? That would work better with the build scripts (or we can add the flag to them as well)